### PR TITLE
【Hackathon 9th No.124】 feat: implement torch._C._nn.avg_pool2d to stable API conversion

### PR DESCRIPTION
**Pull request by Dayuxiaoshui**
### PR Category
Feature Enhancement

### Description
Implement conversion from unstable API torch._C._nn.avg_pool2d to stable API torch.nn.functional.avg_pool2d.

Key changes:
1. Added avg_pool2d_to_avg_pool2d method in UnstableToStableBackend class
   - Traverse FX graph nodes and replace torch._C._nn.avg_pool2d with torch.nn.functional.avg_pool2d
   - Update graph node target function and recompile
   - Replace API calls in generated code string to ensure check_unstable_api can verify correctly

2. Updated unstable_to_stable method
   - Automatically call corresponding conversion function based on DISALLOWED_UNSTABLE_API environment variable

3. Enhanced check_unstable_api method
   - Support using converted code string for verification to ensure accurate API conversion validation

Verification results:
- Tested 149 models, all passed verification
- ES(-6) = 0.9729 (97.29%), far exceeding the requirement of 0.63 (63%)
- All models successfully converted unstable API to stable API

### CODE REVIEW

**Comment by lixinqi** [comment]
这个工作非常棒。再补充一下ES(t)图像到本pr对话里。我们就可以合入了

**Comment by Dayuxiaoshui** [comment]
屏幕截图 2025-11-04 163229
添加了

**Comment by Dayuxiaoshui** [comment]
屏幕截图 2025-11-05 114905
在低容忍度（t < 3）下，ES 值被惩罚为 0.1
在高容忍度（t >= 3）下，ES 值被豁免为 1
当 t < 3 时：fake_perf_degrad 返回 fpdb = 0.1（默认值）
当 t >= 3 时：fake_perf_degrad 对于非 "accuracy" 错误返回 1,可能需要修复

File: graph_net/torch/backend/unstable_to_stable_backend.py [hunk 1]
```diff
@@ -29,8 +29,46 @@ def my_backend(gm, sample_inputs):
...
+        # the generated code will still show torch._C._nn.avg_pool2d
+        # So we need to replace it in the code string
+        code = gm.code
+        modified_code = re.sub(
+            r"torch\._C\._nn\.avg_pool2d\(",
+            "torch.nn.functional.avg_pool2d(",
+            code,
+        )
+        # Store modified code for check_unstable_api to use
+        gm._code_for_check = modified_code
```
Comment (lixinqi) [inline graph_net/torch/backend/unstable_to_stable_backend.py]: 这个地方的改动应该会被其他地方用到才行。应该新添加一个fx_graph_serialize_util.py，在其中添加序列化函数：
```python
def serialize_graph_module_to_str(gm: fx.GraphModule):
        code = gm.code
        code = re.sub(
            r"torch\._C\._nn\.avg_pool2d\(",
            "torch.nn.functional.avg_pool2d(",
            code,
        )
        return code
```
这个api首先应该在下边的check_unstable_api函数里被调用，另外还需要在https://github.com/PaddlePaddle/GraphNet/blob/develop/graph_net/torch/extractor.py#L92 和https://github.com/PaddlePaddle/GraphNet/blob/develop/graph_net/torch/extractor.py#L94 这两处位置被调用

**Comment by Dayuxiaoshui** [comment]
屏幕截图 2025-11-05 164118
最新生成图，完成后

File: graph_net/torch/backend/unstable_to_stable_backend.py (hunk 1 above)
Comment (Dayuxiaoshui) [reply graph_net/torch/backend/unstable_to_stable_backend.py]: 修复完成

**Review by lixinqi — APPROVED** [review]

File: graph_net/torch/backend/unstable_to_stable_backend.py [hunk 2]
```diff
@@ -29,8 +30,32 @@ def my_backend(gm, sample_inputs):
...
+        # Update graph nodes: replace torch._C._nn.avg_pool2d with F.avg_pool2d
+        for node in gm.graph.nodes:
+            if node.op == "call_function":
+                if (
+                    hasattr(node.target, "__module__")
+                    and hasattr(node.target, "__name__")
+                    and node.target.__module__ == "torch._C._nn"
+                    and node.target.__name__ == "avg_pool2d"
+                ):
+                    node.target = F.avg_pool2d
```
Comment (lixinqi) [inline graph_net/torch/backend/unstable_to_stable_backend.py:48]: 我比较讨厌嵌套太深，会习惯性地改为如下列表解析的形式：
```python
issue_nodes = (
    node
    for node in gm.graph.nodes
    if node.op == "call_function"
    if hasattr(node.target, "__module__")
    if node.target.__module__ == "torch._C._nn"
    if hasattr(node.target, "__name__")
    if node.target.__name__ == "avg_pool2d"
)
for node in issue_nodes:
    node.target = F.avg_pool2d
```

Comment (Dayuxiaoshui) [reply graph_net/torch/backend/unstable_to_stable_backend.py:48]: 好滴好滴，收到
